Add lsm6dsl specific samples + change its dts binding

avisconti opened the pull request:
This PR is composed by 3 commits, all 3 somehow related to lsm6dsl sensor.

avisconti committed:
boards/arm: disco_l475_iot1: Add lsm6dsl int gpio info in dts

Provide through the dts the gpio on which the lsm6dsl INT1 is
connected. Enable also the lsm6dsl trigger mode.

Signed-off-by: Armando Visconti <armando.visconti@st.com>

avisconti committed:
samples: sensor: lsm6dsl

This commit provides sample application for sensor lsm6dsl.
This sample has been tested on both 96b_argonkey board,
where lsm6dsl is connect to the micro thru SPI bus, and on
disco_l475_iot1 board, where instead it is connected to I2C bus.

Signed-off-by: Armando Visconti <armando.visconti@st.com>

avisconti commented:
@galak @MaureenHelm 
If you can also provide your inputs would be good!

dbkinder commented on samples/sensor/lsm6dsl/README.rst:
```diff
@@ -0,0 +1,64 @@
+References
+**********
+
+ - LSM6DSL http://www.st.com/en/mems-and-sensors/lsm6dsl.html
```
Rendering is better without the leading space:
```suggestion
- LSM6DSL http://www.st.com/en/mems-and-sensors/lsm6dsl.html
```

avisconti replied:
ack

dbkinder reviewed (changes requested):
Thanks for the documentation, one little tweak.

erwango reviewed (commented):
Seems in conflict with #11854. Can you have a check?

avisconti commented:
@erwango 
> Seems in conflict with #11854. Can you have a check?

Yes, I already noticed it, as @galak added me in the review of #11854.
Actually, I have also to change the commit (just meged) for lis2dh as well have the one for lis3dh (currently under review #11814)

avisconti committed:
dts: sensor: lsm6dsl-i2c: Add irq-gpios property

Add the irq-gpios property to LSM6DSL sensor yaml description
file for I2C case.

Signed-off-by: Armando Visconti <armando.visconti@st.com>

avisconti commented:
I have repushed it on top of #11854. Now both i2c and spi yaml file are using same compatible
string. Moreover I took care of the inputs from @dbkinder on README doc file.

galak commented:
Going to mark this DNM for now until #11854 goes in.

dbkinder reviewed (approved):
doc changes LGTM, thanks!

galak commented:
Can you rebase now that we merged #11854

avisconti commented:
> Can you rebase now that we merged #11854

done!

avisconti commented:
recheck

avisconti commented:
Am I the only one who is getting this error during CI?

---------------
L/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/out-3nd-pass/disco_l475_iot1/samples/net/wifi/test/zephyr -lgcc -Wl,--print-memory-usage -mthumb -nostdlib -static -no-pie --coverage -Wl,-X -Wl,-N -Wl,--gc-sections -Wl,--build-id=none -Wl,--orphan-handling=warn -mabi=aapcs && :
Memory region         Used Size  Region Size  %age Used
           FLASH:       81788 B         1 MB      7.80%
            SRAM:       32640 B        96 KB     33.20%
        IDT_LIST:         184 B         2 KB      8.98zephyr/libzephyr.a(eswifi_bus_spi.c.obj):(.data.eswifi_bus_ops_spi+0x0): multiple definition of `eswifi_bus_ops_spi'
zephyr/libzephyr.a(eswifi_core.c.obj):(.bss.eswifi_bus_ops_spi+0x0): first defined here
zephyr/libzephyr.a(eswifi_bus_spi.c.obj): In function `eswifi_spi_init':
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/drivers/wifi/eswifi/eswifi_bus_spi.c:101: multiple definition of `uarthost'
zephyr/libzephyr.a(eswifi_core.c.obj):/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/drivers/wifi/eswifi/eswifi_core.c:294: first defined here
zephyr/libzephyr.a(eswifi_offload.c.obj):(.bss.eswifi_bus_ops_spi+0x0): multiple definition of `eswifi_bus_ops_spi'
zephyr/libzephyr.a(eswifi_core.c.obj):/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/drivers/wifi/eswifi/eswifi_core.c:294: first defined here
zephyr/libzephyr.a(eswifi_offload.c.obj): In function `eswifi_off_accept':
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/drivers/wifi/eswifi/eswifi_offload.c:88: multiple definition of `uarthost'
zephyr/libzephyr.a(eswifi_core.c.obj):/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/drivers/wifi/eswifi/eswifi_core.c:294: first defined here
collect2: error: ld returned 1 exit status
%
ninja: build stopped: subcommand failed.
-----------------------------------

galak commented:
> @galak
> Am I the only one who is getting this error during CI?

Nope, you aren't.  Looking to see how this happened.

galak commented:
> Nope, you aren't. Looking to see how this happened.

Fix in PR #11988

avisconti commented:
> Fix in PR #11988 

OK, I'll wait for the merge of that PR

avisconti commented:
Merged!

galak reviewed (commented):
Just noticed, and while not directly related CONFIG_LSM6DSL_BUS_TYPE we should be able to replace with {DT_ST_LSM6DSL_BUS_I2C and/or DT_ST_LSM6DSL_BUS_SPI} that are now generated.

galak reviewed (changes requested):
The sample never gets built for an platform via sanitycheck:

./scripts/sanitycheck -T samples/sensor/lsm6dsl/

I wondering if we can wait for #11800 and than use a filter on DT_COMPAT_ST_LSM6DSL

avisconti commented:
This PR has been rebased onto #12105 
Must wait til that one will be merged

avisconti commented:
rebased after #12105 merging.
@galak 
I added DT_COMPAT_ST_LSM6DSL filter in sample.yaml, but still it seems sanitycheck is never
called for any platform. I'm not so familiar with this scripts, so I may need an help from you.

galak reviewed (changes requested):
You need to add 'lsm6dsl' to the board yaml that support it.

avisconti committed:
boards/arm: 96b_argonkey: add supported sensors to yaml file

Add supported sensor to yaml file.
The ArgonKey board supports following sensors:
	- HTS221 (humidity)
	- LPS22HB (pressure)
	- LSM6DSL (imu)
	- VL53L0X (proximity)

Signed-off-by: Armando Visconti <armando.visconti@st.com>

avisconti commented:
@galak 
> you need to add 'lsm6dsl' to the board yaml that support it.

removed the filter and added lsm6dsl to ArgonKey board.
Then re-based and re-pushed!
Thx for your support.

erwango reviewed (changes requested):
2 minor comments, otherwise tested ok

avisconti commented on boards/arm/96b_argonkey/96b_argonkey.yaml:
```diff
@@ -14,4 +14,7 @@ supported:
   - hts221
   - rtc
   - spi
+  - lsm6dsl
```
Sure.

avisconti committed:
boards/arm: disco_l475_iot1: add supported sensors to yaml file

Add LPS22HB and LSM6DSL sensors to yaml file.

Signed-off-by: Armando Visconti <armando.visconti@st.com>

avisconti commented on samples/sensor/lsm6dsl/src/main.c:
```diff
@@ -0,0 +1,171 @@
+	}
+
+	while (1) {
+		/* lsm6dsl accel */
```
ack

avisconti commented:
Modified according to @erwango observation and re-pushed.
